tests/foreman/api/test_template.py

```diff
@@ -306,3 +312,334 @@ def test_positive_clone(self):
+            5. Administer -> Settings -> TemplateSync settings must be set
+               for default options:
+               branch -> develop
+               repo -> e.g: ssh://git@github.com/username/community-templates
+               prefix - > robottelo (or something else easy to test)
+            6. The git repository must be populated with some dummy templates
+               for testing and that repo must have content in different
+               branches and diverse directory roots. Based on
+               foreman/community templates create some templates with
+               diverse set of names and prefixes to use for testing the
+               regex filtering and also some templates must have Org/Loc
+               metadata matching the previously created Org/Loc. the process of
+               population this templates repository can be part of this setUp.
+
+               e.g: https://github.com/theforeman/community-templates/tree/
+                    develop/provisioning_templates/provision
+                    in the above directory there are templates prefixed with
+                    `alterator` `atomic` `coreos` `freebsd` etc..
+
+        :info:
+            - https://theforeman.org/plugins/foreman_templates/5.0/index.html
+            - /apidoc/v2/template/import.html
+            - /apidoc/v2/template/export.html
+
+        """
+
+    # Import tests
+
+    @stubbed()
+    @tier1
+    def test_positive_import_filtered_templates_from_git(self):
+        """Assure only templates with a given filter regex are pulled from
+        git repo.
+
+        :id: 628a95d6-7a4e-4e56-ad7b-d9fecd34f765
+
+        :Steps:
+            1. Using nailgun or direct API call
+               import only the templates matching with regex e.g: `^atomic.*`
+               refer to: `/apidoc/v2/template/import.html`
+
+        :expectedresults:
+            1. Assert result is {'message': 'success'} and template imported
+            2. Assert no other template has been imported but only those
+               matching specified regex.
+               NOTE: Templates are always imported with a prefix defaults to
+               `community` unless it is specified as empty string
+
+        :CaseImportance: Critical
+        """
+
+    @stubbed()
+    @tier1
+    def test_negative_import_filtered_templates_from_git(self):
+        """Assure templates with a given filter regex are NOT pulled from
+        git repo.
+
+        :id: a6857454-249b-4a2e-9b53-b5d7b4eb34e3
+
+        :Steps:
+            1. Using nailgun or direct API call
+               import the templates NOT matching with regex e.g: `^freebsd.*`
+               refer to: `/apidoc/v2/template/import.html` using the
+               {'negate': true} in POST body to negate the filter regex.
+
+        :expectedresults:
+            1. Assert result is {'message': 'success'}
+            2. Assert templates mathing the regex were not pulled.
+
+        :CaseImportance: Critical
```

Comment:
we need to revisit once all tests to see if every test is really `critical` ?

Reply:
@sghai All template sync operations (import & export) will overwrite existing templates, so it is **critical** to have things like filters and subdirectories working.

Reply:
cool.. thanks